python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -0,0 +1,263 @@
+        """
+        if not self.isDefined(self.optimizer):
+            raise ValueError("Required Param optimizer must be specified but was not.")
+        if not is_valid_optimizer(self.getOptimizer()):
+            raise ValueError("Named optimizer not supported in Keras: %s" % self.getOptimizer())
+        if not self.isDefined(self.loss):
+            raise ValueError("Required Param loss must be specified but was not.")
+        if not is_valid_loss_function(self.getLoss()):
+            raise ValueError("Named loss not supported in Keras: %s" % self.getLoss())        
+        if not self.isDefined(self.inputCol):
+            raise ValueError("Input column must be defined")
+        if not self.isDefined(self.outputCol):
+            raise ValueError("Output column must be defined")
+        return True
+
+    def _localFit(self, featuresBc, labelsBc, modelBytesBc, paramMap):
+        # Copy the estimator to add to the Params without modifying this instance.        
+        estimator = self.copy(paramMap)
+        features = featuresBc.value
+        labels = None if labelsBc is None else labelsBc.value
+        model = bytes_to_model(modelBytesBc.value)
+
+        model.compile(optimizer=estimator.getOptimizer(), loss=estimator.getLoss())
+        fit_params = estimator.getKerasFitParams()
+
+        model.fit(x=features, y=labels, **fit_params)
+        return model
+
+    def _getNumpyFeaturesAndLabels(self, dataset):
+        image_uri_col = self.getInputCol()
+        label_col = None
+        if self.isDefined(self.labelCol) and self.getLabelCol() != "":
+            label_col = self.getLabelCol()
+        tmp_image_col = self._loadedImageCol()        
+        image_df = self.loadImagesInternal(dataset, image_uri_col).dropna(subset=[tmp_image_col])
+        
+        localFeatures = []
+        for row in image_df.select(tmp_image_col).rdd.toLocalIterator():
+            spimg = row.asDict()[tmp_image_col]
+            features = imageStructToArray(spimg)
+            localFeatures.append(features)
+            
+        if len(localFeatures) == 0:
+            raise ValueError("Given empty dataset!")        
+
+        # We must reshape input data to form to the same size so as to be stacked
+        logger.warn('user provided imageLoader must convert all images to the same size')
+        X = np.stack(localFeatures, axis=0)
+
+        if label_col is None:
+            return X, None
+
+        localLabels = []
+        for row in image_df.select(label_col).rdd.toLocalIterator():
+            localLabels.append(int(row.label))
+        if len(localLabels) != len(localFeatures):
+            raise ValueError("Number of labels {} not matching number of instances {}"
+                             .format(len(localLabels), len(localFeatures)))
+        # We also need to convert them to one-hot encoding
+        y_category = np.stack(localLabels, axis=0)
+        batch_size = y_category.shape[0]
```

Comment:
The use of `stack` and `shape` is confusing here because it makes it seem like `y_category` is multidimensional. Could we do something like:

```
localLabel = np.asarray(localLabel)
batch_size = len(localLabel)
```